src/Map.jl

```diff
@@ -0,0 +1,221 @@
+abstract type AbstractMap{D,CoD} end
+
+domain(m::AbstractMap) = m.domain
+
+codomain(m::AbstractMap) = m.codomain
+
+struct CompositeMap{D,CoD,F,G}
+    f::F
+    g::G
+end
+
+function CompositeMap(f::F, g::G) where {D,
+                                         I,
+                                         CoD,
+                                         F<:AbstractMap{I,CoD},
+                                         G<:AbstractMap{D,I}}
```

Comment:
Why do you require codomain of `G` to be exactly equal to domain of `F`? I think it's enough if that codomain is a subset of domain of `F`.

Reply:
Largely for clarity. One could always use an inclusion map to bridge the gap, and since that would basically just be the identity function, it wouldn't add any overhead. Ultimately, these maps should be able to take `MPoint`s, `TVector`s, and such, and a custom `MPoint` for a submanifold won't necessarily be an `MPoint` for its parent manifold without an inclusion map, whose function might be to change `MPoint` types.